Services/Ping/NationPingService.cs

```diff
@@ -29,45 +32,45 @@ protected override Task ExecuteAsync(CancellationToken cancellationToken)
             _logger.LogInformation(
                 "Ping Service listening on [{ServiceOptionsListenAddress}:{ServiceOptionsListenPort}]",
                 _options.ListenAddress, _options.ListenPort);
-            return RequestHandler(cancellationToken);
+
+            return ProcessAccept(cancellationToken);
         }
 
-        private async Task RequestHandler(CancellationToken cancellationToken)
+        private async Task ProcessAccept(CancellationToken cancellationToken)
         {
             while (!cancellationToken.IsCancellationRequested)
             {
-                var client = await _tcpListener.AcceptTcpClientAsync(cancellationToken).ConfigureAwait(false);
-                var innerCancellationToken = new CancellationTokenSource();
-                var linkedCancellationToken = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, innerCancellationToken.Token);
-                try
-                {
-                    await DoRequest(client, linkedCancellationToken.Token);
-                }
-                catch
-                {
-                    innerCancellationToken.Cancel();
-                }
+                var socket = await _tcpListener.AcceptSocketAsync(cancellationToken).ConfigureAwait(false);
+                if (_logger.IsEnabled(LogLevel.Debug))
```

Comment:
I'd rather let the Log Service decide if it wants to print or not, so I'd leave this conditional check out
```suggestion
                if (_logger.IsEnabled(LogLevel.Debug))
```